adapter/memory/src/main/kotlin/com/xorker/draw/mafia/MafiaGameWaitingQueueAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package com.xorker.draw.mafia
+
+import com.xorker.draw.exception.UnSupportedException
+import com.xorker.draw.mafia.event.MafiaGameRandomMatchingEvent
+import com.xorker.draw.websocket.WaitingQueueSession
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedQueue
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+
+@Component
+internal class MafiaGameWaitingQueueAdapter(
+    private val eventPublisher: ApplicationEventPublisher,
+) : MafiaGameWaitingQueueRepository {
+    private val koWaitingQueue: ConcurrentLinkedQueue<WaitingQueueSession> = ConcurrentLinkedQueue()
+    private val enWaitingQueue: ConcurrentLinkedQueue<WaitingQueueSession> = ConcurrentLinkedQueue()
+    private val waitingQueue: ConcurrentHashMap<String, ConcurrentLinkedQueue<WaitingQueueSession>> = ConcurrentHashMap()
+    private val lock = Object()
+
+    init {
+        waitingQueue["ko"] = koWaitingQueue
+        waitingQueue["en"] = enWaitingQueue
+    }
+
+    override fun enqueue(size: Int, session: WaitingQueueSession) {
+        val queue = waitingQueue[session.locale] ?: throw UnSupportedException
```

Comment:
p4. 언어가 추가 될 때 마다 초기화 해주기 보다는 조회 할 때 초기화 해주면 대응이 유연할 것 같아요

```suggestion
    override fun enqueue(size: Int, session: WaitingQueueSession) {
        val queue = waitingQueue.getOrPut(session.locale) { ConcurrentLinkedQueue() }
```